webui/react/src/components/InlineEditor.test.tsx

```diff
@@ -89,9 +89,28 @@ describe('InlineEditor', () => {
     expect(onCancel).toHaveBeenCalled();
   });
 
-  it('doesnt allow user input when disabled', async () => {
-    setup({ disabled: true, onSaveReturnsError: true, value: 'before' });
+  it('should not allow user input when disabled', async () => {
+    const { user } = setup({ disabled: true, onSaveReturnsError: true, value: 'before' });
     await user.type(screen.getByRole('textbox'), 'after');
     expect(screen.getByDisplayValue('before')).toBeInTheDocument();
   });
+
+  it('should ignore keydown event until the IME is confirmed', async () => {
+    const { waitForSpinnerToDisappear, user } = setup();
+    const textbox = screen.getByRole('textbox');
+    await user.click(textbox);
+    await user.clear(textbox);
+    await user.type(textbox, 'こんにちは');
+    fireEvent.keyDown(textbox, {
+      charCode: 229,
+      code: 'Enter',
+      key: 'Enter',
+      keyCode: 229,
+    });
```

Comment:
I dont wanna use `fireEvent` here but seems like userEvent doesnt have option to specify keyCode

Reply:
yeah, totally get it. I have encountered something like this where fireEvent still had to be used